src/client/mod.rs

```diff
@@ -128,7 +128,7 @@ impl<P: Provider> Client<P> {
         ]);
         let body = body.finish();
 
-        let mut response = http_client.post(self.provider.token_uri())?
+        let mut response = http_client.post(self.provider.token_uri().as_str())?
```

Comment:
Does this mean `http_client` re-parses the URL?

Reply:
Good catch, you can avoid the reparse by cloning Url - reqwest impls `IntoUrl` for its methods only for owned Urls. I committed that change.